src/monodroid/monodroid.targets

```diff
@@ -22,22 +19,33 @@
         Overwrite="True"
     />
     <Exec
-        Command="$(AndroidToolchainDirectory)\ndk\ndk-build CONFIGURATION=$(Configuration) V=1"
+        Command="$(AndroidToolchainDirectory)\ndk\ndk-build CONFIGURATION=Debug V=1"
     />
     <Copy
         SourceFiles="@(AndroidSupportedTargetJitAbi->'obj\local\%(Identity)\libmonodroid.so')"
-        DestinationFiles="@(AndroidSupportedTargetJitAbi->'$(OutputPath)\%(Identity)\libmono-android.$(_Conf).d.so')"
+        DestinationFiles="@(AndroidSupportedTargetJitAbi->'$(OutputPath)\%(Identity)\libmono-android.debug.d.so')"
     />
     <Copy
         SourceFiles="@(AndroidSupportedTargetJitAbi->'libs\%(Identity)\libmonodroid.so')"
-        DestinationFiles="@(AndroidSupportedTargetJitAbi->'$(OutputPath)\%(Identity)\libmono-android.$(_Conf).so')"
+        DestinationFiles="@(AndroidSupportedTargetJitAbi->'$(OutputPath)\%(Identity)\libmono-android.debug.so')"
+    />
+    <Exec
+        Command="$(AndroidToolchainDirectory)\ndk\ndk-build CONFIGURATION=Release V=1"
```

Comment:
That doesn't answer the question: Do we need to "clean" to ensure that things are rebuilt? To rephrase it, what happens when you do:

    $ ndk-build CONFIGURATION=Debug V=1
    $ ndk-build CONFIGURATION=Release V=1

...since that's what this PR *does*.

*For me*, it does...nothing:

    $ ndk-build CONFIGURATION=Debug
    [armeabi-v7a] Install        : libmonodroid.so => libs/armeabi-v7a/libmonodroid.so
    $ ndk-build CONFIGURATION=Release
    [armeabi-v7a] Install        : libmonodroid.so => libs/armeabi-v7a/libmonodroid.so

Changing `CONFIGURATION` does *not* cause `ndk-build` to rebuild *anything* if the outputs are already up-to-date.

Meaning your "Release" *won't actually be a release build!*

(This is *probably* why it even builds for you *at all*.)

To force a rebuild with configuration change, an intermediate clean is required:

    $ ndk-build CONFIGURATION=Debug
    ...
    $ ndk-build clean
    $ ndk-build CONFIGURATION=Release

...which *fails* for me:

    In file included from jni/../../../external/mono/support/nl.h:3:0,
                     from jni/../../../external/mono/support/nl.c:14:
    jni/../../../external/mono/eglib/src/glib.h:22:26: fatal error: eglib-config.h: No such file or directory
     #include <eglib-config.h>
                              ^
    compilation terminated.
    make: *** [obj/local/armeabi-v7a/objs/monodroid/__/__/__/external/mono/support/nl.o] Error 1

Which makes sense: it's trying to use `$(CONFIGURATION)` to grab `eglib-config`, from `$(LOCAL_PATH)/../../../build-tools/mono-runtimes/obj/$(CONFIGURATION)/$(TARGET_ARCH_ABI)/eglib/src`, and if we haven't build Mono in the Release configuration -- which we *haven't* -- then this can't possibly work.